1_prep/src/munge_netCDFs.R

```diff
@@ -1,29 +1,155 @@
+               time <= as.Date("2010-01-01") ~ gcm_time_periods[1],
+               time <= as.Date("2070-01-01") ~ gcm_time_periods[2],
+               TRUE ~ gcm_time_periods[3])
+             ) %>%
+      # Reorder the columns to match the standard order
+      select(time_period, time, Shortwave, Longwave, AirTemp, RelHum, WindSpeed, Rain, Snow) %>%
+      # Group by time period
+      group_by(time_period) %>%
+      # For each time period...
+      group_map(~ {
+        time_period <- .y$time_period
+        # add the specified number of burn_in and burn_out days
+        meteo_w_burn_in_out <- add_burn_in_out_to_meteo(.x)
+        # then write the data for the current cell to a csv file
+        outfile <- sprintf(outfile_template, gcm_name, time_period, cell_no)
+        fwrite(setDT(meteo_w_burn_in_out), outfile)
+        return(outfile)
+      })
+  }) %>% unlist()
+
+  return(time_period_cell_files)
+}
 
+#' @title Munge GCM dates
+#' @description Function to determine the start and end dates of each
+#' GCM time period and document the date when burn-in stars and the date
+#' when burn-out ends
+#' @param gcm_nc filenames of the GCM netCDF files
+#' @param gcm_time_periods - the three GCM time periods, defined by their 
+#' bracketing years
+#' @param burn_in length of burn-in period, in days. Used to mirror the 
+#' start of the raw meteorological data to create a burn-in period for 
+#' the model simulation.
+#' @param burn_out length of burn-out period, in days. Used to mirror the 
+#' end of the raw meteorological data to create a burn-out period for 
+#' the model simulation.
+#' @return a tibble with a row for each time period, and columns specifying
+#' the start and end date of each time period, as well as the length of
+#' burn in, the burn-in start date, the length of burn-out, and the burn-out
+#' end date for each time period
+munge_gcm_dates <- function(gcm_time_periods, gcm_ncs, burn_in, burn_out) {
+  # Use first nc file, since all have the same date range
+  nc <- ncdfgeom::read_timeseries_dsg(gcm_ncs[1])
   
-  # write to feather file, specific to gcm, cell, and time period
-  outfile <- sprintf(outfile_template, gcm_name, gcm_date, cell_no)
-  arrow::write_feather(data, outfile)
+  nc_dates <- tibble(
+    time = lubridate::as_date(nc$time),
+    time_period = case_when(
+      time <= as.Date("2010-01-01") ~ gcm_time_periods[1],
+      time <= as.Date("2070-01-01") ~ gcm_time_periods[2],
+      TRUE ~ gcm_time_periods[3])
+    ) %>%
+    group_by(time_period) %>%
+    summarize(gcm_start_date = min(time), gcm_end_date = max(time)) %>%
+    mutate(
+      # a burn-in period only will have been added if the requested burn-in period was 
+      # shorter than the length of the raw meteo data
+      burn_in = case_when(
+        burn_in <= (gcm_end_date-gcm_start_date) ~ burn_in, 
+        TRUE ~ 0),
+      burn_in_start = case_when(
+        burn_in > 0 ~ gcm_start_date-burn_in+1, 
```

Comment:
I thought that too initially, but it proved to not be the case because of how the burn-in/out is used to mirror the data in [Jordan's code](https://github.com/hcorson-dosch/lake-temperature-process-models/blob/read_meteo_from_nc_files/1_prep/src/munge_netCDFs.R#L16-L33) . The # of days added ends up being one less than requested, in order to mirror the data around the actual start/end date of the GCM data.

Reply:
Hmm I guess I could just tweak Jordan's code to account for that loss of a day, which then would mean added burn-in/out = requested burn-in/out

Reply:
I think it's actually all cleaner now with that tweak - since the # days added matches the requested burn-in/out